core/sql/optimizer/RelExpr.cpp

```diff
@@ -9062,42 +9055,19 @@ void Scan::setTableAttributes(CANodeId nodeId)
   setBaseCardinality(MIN_ONE (tableDesc->getNATable()->getEstRowCount())) ;
 }
 
-NABoolean Scan::equalityPredOnCol(ItemExpr *col)
-{
-  //
-  // Returns TRUE if the column (col) has an equality predicate
-  // associated with it the scan's predicate list.
-  //
-  ValueIdSet pred = getSelectionPredicates();
-
-  for (ValueId vid=pred.init(); pred.next(vid); pred.advance(vid))
-  {
-    if ( vid.getItemExpr()->getOperatorType() != ITM_VEG_PREDICATE )
-      continue;
-
-    ItemExpr *expr = vid.getItemExpr();
-    ValueId id;
-    VEG *veg = ((VEGPredicate*)expr)->getVEG();
-
-    if (veg->getAllValues().referencesTheGivenValue(col->getValueId(), id))
-      return TRUE;
-  }
-
-  return FALSE;
-
-} // Scan::equalityPredOnCol
-
 NABoolean Scan::updateableIndex(IndexDesc *idx)
 {
   //
-  // Returns TRUE if the index (idx) can be used for a scan during an UPDATE.
-  // Otherwise, returns FALSE to prevent the "Halloween Update Problem".
+  // Returns TRUE if the index (idx) can be used for a scan during 
+  // an UPDATE. Halloween problem is protected with a sort using
+  // Scan::requiresHalloweenForUpdateUsingIndexScan(). 
+  // Returns FALSE only for certain embedded updates now.
```

Comment:
See also comments below. I wonder why you changed this method, which seemed good to me, to something that no longer corresponds to its name?